### PR TITLE
Phone number validation (libphonenumber-js & MsPhoneNumberInput)

### PR DESCRIPTION
**Pull request by NicoTuxx**
Implementation of https://www.npmjs.com/package/libphonenumber-js#using-phone-number-validation-feature
linked to [parsec-sign#38](https://github.com/Scille/parsec-sign/issues/38)

### CODE REVIEW

File: lib/common/validation/phoneNumber.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { IValidator, ValidationResult, Validity } from '@lib/common/validation';
```
Comment (Max-7) [inline lib/common/validation/phoneNumber.ts:3]: Put those in a `types.ts` and import the file directly, not the module. This creates a circular import, I'm surprised the linter doesn't complain.

File: lib/components/ms-input/MsPhoneNumberInput.vue [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    :placeholder="placeholder"
+    id="ms-phone-input"
+    @change="onChange"
+    v-model="phoneNumber"
+    :validator="phoneNumberValidator"
+  />
+</template>
+
+<script setup lang="ts">
+import { MsInput } from '@lib/components';
```
Comment (Max-7) [inline lib/components/ms-input/MsPhoneNumberInput.vue:15]: Import the full file from inside the module.

File: src/views/HomePage.vue [hunk 3]
```diff
@@ -452,11 +462,22 @@ const toastTheme = ref(MsReportTheme.Success);
...
 const addressInput = ref();
+const phoneNumberInput = ref();
 const VALID_CODE = ['1', '2', '3', '4', '5', '7'];
 const progress = ref(0);
 const stripeCardForm = ref();
 const stripeCardDetails = ref<PaymentMethod.Card>();
 
+const countryCode = computed<CountryCode>(() => {
+  switch (I18n.getLocale()) {
+    case 'fr':
```
Comment (Max-7) [inline src/views/HomePage.vue:473]: `getLocale()` returns `fr-FR` or `en-US`.

**Review by Max-7 — CHANGES_REQUESTED** [review]

**Comment by NicoTuxx** [comment]
PR is suspended for now (other priorities on the project), however to finish it, take a look at this example to fix the usage of the validator:

You can make a closure for this:
```
 const wordValidator = ((word: string): (value: string) => Promise<ValidationResult> => {
   var validator = async function (value: string): Promise<ValidationResult> {
     return { validity: word === value ? Validity.Valid : Validity.Invalid };
   }
   return validator;
 });
```

And use it like this:
` <ms-input :validator="wordValidator('parsec')" />`
Returns Valid if input is strictly equal to 'parsec'.
Note: there seems to be a bug at focus loss, the model seems to reset, no idea why though.

**Review by Max-7 — APPROVED** [review]

**Pull request merged**